serving-api/src/main/scala/com/stratio/sparta/serving/api/service/http/FragmentHttpService.scala

```diff
@@ -173,10 +178,17 @@ trait FragmentHttpService extends BaseHttpService {
       put {
         entity(as[FragmentElementModel]) { fragment =>
           complete {
-            val future = supervisor ? new Update(fragment)
-            Await.result(future, timeout.duration) match {
-              case Response(Success(fragment)) => HttpResponse(StatusCodes.OK)
-              case Response(Failure(exception)) => throw exception
+            val policyActor = actors.get(AkkaConstant.PolicyActor).get
+            val fragmentStatusActor = actors.get(AkkaConstant.FragmentActor).get
+            for {
+              updateResponse <- fragmentStatusActor ? FragmentActor.Update(fragment)
+              relatedPolicies <- policyActor ? PolicyActor.FindByFragment(fragment.fragmentType,
+                fragment.id.getOrElse(throw new RuntimeException("No valid policy id")))
+            } yield (updateResponse, relatedPolicies) match {
```

Comment:
I think that you do not need to yield updateResponse only relatedPolicies will be used in the match